src/Superstructure/ESMFMod/interface/ESMF_Init_C.F90

```diff
@@ -59,14 +59,26 @@ subroutine f_esmf_frameworkinitialize(lang, configFileName, &
 
    end subroutine f_esmf_frameworkinitialize
 
-   subroutine f_esmf_frameworkfinalize(rc)
+   subroutine f_esmf_frameworkfinalize(rc, keepMpi)
        use ESMF_CompMod
        use ESMF_InitMod
+       use ESMF_UtilTypesMod
+       use, intrinsic :: iso_c_binding
 
        implicit none
 
        integer :: rc
+       logical(kind=c_bool), intent(in), optional :: keepMpi
+       type(ESMF_End_Flag) :: endflag
 
-       call ESMF_Finalize(rc=rc)
+       if (present(keepMpi)) then
+         if (keepMpi) then
+            endflag = ESMF_END_KEEPMPI
+         else
+            endflag = ESMF_END_NORMAL
+         endif
+       endif
```

Comment:
It looks like this needs an `else` block for the case when `keepMpi` is absent: `endflag` should be set to `ESMF_END_NORMAL` in that case.

Reply:
Resolved in [7249651](https://github.com/esmf-org/esmf/pull/234/commits/72496513fa15a0d867dd15d0be73dab5f0ad3e3a). Thanks for catching this!